From aca0da7a6a86e3b6dd374b473b6320eef13e5ddf Mon Sep 17 00:00:00 2001
From: "Roy, UjjaL" <uroy@maxlinear.com>
Date: Sun, 6 Jun 2021 08:52:07 +0530
Subject: [PATCH] UGW_SW-59378: Fixed compilation when disabling mcast-snooping

When disabling Lantiq Multicast Snooping support from Linux Kernel,
observed compilation issue due to unblocked code for that feature.

UGW_SW-58765: Re-Engineer Multicast framework.

Signed-off-by: Roy, UjjaL <uroy@maxlinear.com>
---
 net/bridge/br_forward.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/bridge/br_forward.c b/net/bridge/br_forward.c
index f7b1fb4a2972..99dce1de54ca 100644
--- a/net/bridge/br_forward.c
+++ b/net/bridge/br_forward.c
@@ -154,6 +154,7 @@ static int deliver_clone(const struct net_bridge_port *prev,
 		}
 	}
 
+#ifdef CONFIG_MCAST_SNOOPING
 	/* hook the UDP multicast to mcast helper */
 	if (bridge_lanserver_hook == 1) {
 		if (!is_broadcast_ether_addr(dest) && is_multicast_ether_addr(dest) && ((ip_hdr(skb)->protocol == IPPROTO_UDP) || (ipv6_hdr(skb)->nexthdr == IPPROTO_UDP))) {
@@ -164,6 +165,7 @@ static int deliver_clone(const struct net_bridge_port *prev,
 			}
 		}
 	}
+#endif
 #endif
 
 	__br_forward(prev, skb, local_orig);
